File: vscodes/canJump.cpp
#include <iostream>
#include <vector>
using namespace std;

/**
 * 55. 跳跃游戏
 * */
bool canJumps(vector<int> vector);

using namespace std;
int main() {
//    Solution solution;
    vector<int> nums{0,3,1,1,4};
    bool d = canJumps(nums);
    cout<< d << endl;
    return 0;
}

bool canJumps(vector<int> nums)
{
    int k = 0;
    for (int i = 0; i < nums.size(); i++)
    {
        if (i > k)
            return false;
        k = max(k, i + nums[i]);
    }
    return true;
}

File: vscodes/CanJump.java
public class CanJump {
    public boolean canJump(int[] nums){
        for (int i = 0;i<nums.length;i++) {
            if(nums[i]==0)
                if(passZero(nums,i))
                    continue;
                else
                    return false;
        }
        return true;
    }
    //判断是否能跳出当前0
    public static boolean passZero(int[] nums,int i) {
        for(int j=0;j<i;j++) { // 在 i 之前找到能跳过 0 的下标
            if(j+nums[i]>i)
                return true;
        }
        return false;
    }
}